src/read.cpp

```diff
@@ -46,6 +46,9 @@ CharacterVector read_lines_(List sourceSpec, List locale_,
 Function R6method(Environment env, const std::string& method) {
   return as<Function>(env[method]);
 }
+bool isTrue(SEXP x) {
+  return TYPEOF(x) == LGLSXP && Rf_length(x) == 1 && LOGICAL(x)[0] == TRUE;
```

Comment:
I wonder if we should throw an explicit error if the output is the wrong type (i.e. not length 1 logical vector)